components/odh-notebook-controller/controllers/notebook_webhook_test.go

```diff
@@ -165,6 +162,62 @@ var _ = Describe("The Openshift Notebook webhook", func() {
 				// there is no update to the Notebook
 				expectedImage: ":some-tag",
 				expectedEvents: []string{
+					IMAGE_STREAM_TAG_NOT_FOUND_EVENT,
+				},
+			},
+			{
+				name: "ImageStream in the same namespace as the Notebook",
+				imageStream: &imagev1.ImageStream{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ImageStream",
+						APIVersion: "image.openshift.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-image",
+						Namespace: Namespace,
+					},
+					Spec: imagev1.ImageStreamSpec{
+						LookupPolicy: imagev1.ImageLookupPolicy{
+							Local: true,
+						},
+					},
+					Status: imagev1.ImageStreamStatus{
+						Tags: []imagev1.NamedTagEventList{{
+							Tag: "some-tag",
+							Items: []imagev1.TagEvent{{
+								Created:              toMetav1Time("2025-05-14T02:36:21Z"),
+								DockerImageReference: "quay.io/modh/odh-generic-data-science-notebook@sha256:5999547f847ca841fe067ff84e2972d2cbae598066c2418e236448e115c1728e",
+								Image:                "sha256:5999547f847ca841fe067ff84e2972d2cbae598066c2418e236448e115c1728e",
+								Generation:           2,
+							}},
+							Conditions: []imagev1.TagEventCondition{{
+								Type:               "ImportSuccess",
+								Status:             "False",
+								LastTransitionTime: toMetav1Time("2025-05-14T02:36:21Z"),
+								Reason:             "NotFound",
+							}},
+						}},
+					},
+				},
+				notebook: &nbv1.Notebook{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      Name,
+						Namespace: Namespace,
+						Annotations: map[string]string{
+							"notebooks.opendatahub.io/last-image-selection": "some-image:some-tag",
+						},
+					},
+					Spec: nbv1.NotebookSpec{
+						Template: nbv1.NotebookTemplateSpec{
+							Spec: corev1.PodSpec{Containers: []corev1.Container{{
+								Name:  Name,
+								Image: ":some-tag",
+							}}}},
+					},
+				},
+				// there is no update to the Notebook
+				expectedImage: "quay.io/modh/odh-generic-data-science-notebook@sha256:5999547f847ca841fe067ff84e2972d2cbae598066c2418e236448e115c1728e",
```

Comment:
Comment above this line is wrong. Ofc there is change to image, it used to be :some-tag and now is quay.io/...

Reply:
thanks for sharing, would be more careful next time :+1: